### DISCUSSION TITLE
Serverless function as middleware

### DISCUSSION THREAD

**Discussion by alexander-r-jaimy** [vercel/vercel#4416] [Help]
Good Morning,

I tried to implement a simple way to intercept a request to my test site with serverless function like this:

`module.exports = (req, res) => {
  return res;
};`

and my function configuration in `vercel.json` file is like this:

`{
  "functions": {
    "api/lambda.js": {
      "memory": 1024
    }
  },
  "routes": [{ "src": "/(.*)", "dest": "/api/handler.js" }]
}
`
Current Behavior:
Unfortunately, a request is not handled further on to an application and after 10 seconds I receive timeout.

Expected Behavior:
I'd like to use a serverless function as middleware between client request and my app.
Possible use cases: basic auth, redirects.

Does someone one khow to achieve this?

**Comment by mcsdevv** [comment]
Hello @alexander-r-jaimy, a really good example of using Serverless Function Middleware can be found here: https://github.com/zeit/next.js/tree/canary/examples/api-routes-middleware

There's no need for any kind of configuration.

**Comment by alexander-r-jaimy** [comment]
Thank you for your response however the problem maintains.
I'm trying another solution which implies running getInitialProps:
![Screenshot from 2020-05-19 13-30-11](https://user-images.githubusercontent.com/60174868/82327750-240fe680-99df-11ea-84cb-e0e541bad8df.png)
However when accessing the link that should be redirected, I receive 404 by server:
![Screenshot from 2020-05-19 13-25-48](https://user-images.githubusercontent.com/60174868/82327796-3558f300-99df-11ea-8dab-6ab804abe133.png)
But getInitialProps functions seems to be returning correct status code
![Screenshot from 2020-05-19 13-25-23](https://user-images.githubusercontent.com/60174868/82327820-40ac1e80-99df-11ea-8d3b-fa9772d16587.png)

I tried the solution as in:
https://github.com/zeit/next.js/issues/649

Looks like it doesn't work anymore.
What am I missing?

**Reply by coetry** [reply]
@alexander-r-jaimy Do you have a public repo that we can look at? You can intercept the client request on a page-by-page basis by calling `getInitialProps` on a specific page instead of at your application root.
